app/controllers/concerns/application_shared.rb

```diff
@@ -4,6 +4,7 @@ module ApplicationShared
   include Foreman::Controller::MigrationChecker
   include Foreman::Controller::Authentication
   include Foreman::Controller::Session
+  include Foreman::Controller::Flash
```

Comment:
shared is also used by api base controller, maybe move it to the application controller instead?